chain/src/chain.rs

```diff
@@ -375,16 +365,22 @@ impl ChainService {
 
             self.shared.store_snapshot(Arc::clone(&new_snapshot));
 
-            tx_pool.update_tx_pool_for_reorg(
-                fork.detached_blocks().iter(),
-                fork.attached_blocks().iter(),
-                fork.detached_proposal_id().iter(),
-                &mut txs_verify_cache,
+            if let Err(e) = self.shared.tx_pool_controller().update_tx_pool_for_reorg(
+                fork.detached_blocks().clone(),
+                fork.attached_blocks().clone(),
+                fork.detached_proposal_id().clone(),
                 new_snapshot,
-            );
+            ) {
+                error!("notify update_tx_pool_for_reorg error {}", e);
```

Comment:
Can we return errors or just use `.expect`? IMO the code should not continue to run after this, it will cause inconsistent silently.

Reply:
The current implementation, notify error can be tolerated. 
TxPool can work asynchronously with chain. 
This error only occurs in high pressure, the channel is full that can't send messages anymore, in this situation, I'm afraid such retry does not any help, so I think just log is fine.
Also, we don't have to interrupt the program immediately, it won't break anything, chain data persistence already finished, maybe we can do more such persistent TxPool, someday.